src/main/scala/mesosphere/marathon/core/condition/Condition.scala

```diff
@@ -49,6 +49,12 @@ object Condition {
   sealed trait Failure extends Terminal
   sealed trait Active extends Condition
 
+  /** Scheduled: Task should be launched by matching offers. Mesos does not know anything about it. */
+  case object Scheduled extends Condition
```

Comment:
what do you think about introducing `sealed trait Initial` for these two (similarly as we have Terminal or Active). Then we can filter instances based on this condition. Filtering based on agentInfo being Some (not None) seems to me that the client has to know too much implementation detail.

Reply:
Good point. However, we do have to filter for the agent info since a lot of methods require the agent info being present and I want to avoid an option get.